tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java

```diff
@@ -0,0 +1,313 @@
+package com.maxio.advancedbilling.controllers.events;
+
+import com.maxio.advancedbilling.AdvancedBillingClient;
+import com.maxio.advancedbilling.TestClientProvider;
+import com.maxio.advancedbilling.exceptions.ApiException;
+import com.maxio.advancedbilling.models.Component;
+import com.maxio.advancedbilling.models.CreateUsage;
+import com.maxio.advancedbilling.models.CreateUsageRequest;
+import com.maxio.advancedbilling.models.CreditCardAttributes;
+import com.maxio.advancedbilling.models.Customer;
+import com.maxio.advancedbilling.models.Direction;
+import com.maxio.advancedbilling.models.Event;
+import com.maxio.advancedbilling.models.EventKey;
+import com.maxio.advancedbilling.models.EventResponse;
+import com.maxio.advancedbilling.models.InvoiceIssued;
+import com.maxio.advancedbilling.models.ItemPricePointChanged;
+import com.maxio.advancedbilling.models.ListEventsInput;
+import com.maxio.advancedbilling.models.MeteredUsage;
+import com.maxio.advancedbilling.models.PaymentRelatedEvents;
+import com.maxio.advancedbilling.models.Product;
+import com.maxio.advancedbilling.models.ProductFamily;
+import com.maxio.advancedbilling.models.Subscription;
+import com.maxio.advancedbilling.models.SubscriptionProductChange;
+import com.maxio.advancedbilling.models.SubscriptionStateChange;
+import com.maxio.advancedbilling.models.UpdateCustomer;
+import com.maxio.advancedbilling.models.UpdateCustomerRequest;
+import com.maxio.advancedbilling.models.UpdateSubscription;
+import com.maxio.advancedbilling.models.UpdateSubscriptionRequest;
+import com.maxio.advancedbilling.models.containers.CreateUsageComponentId;
+import com.maxio.advancedbilling.utils.TestSetup;
+import com.maxio.advancedbilling.utils.matchers.EventSpecificDataGetter;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.maxio.advancedbilling.models.EventKey.*;
+import static org.apache.commons.lang3.RandomStringUtils.randomNumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class EventsControllerListEventsTest {
+
+    private final AdvancedBillingClient client = TestClientProvider.getClient();
+    private final TestSetup testSetup = new TestSetup();
+
+    ProductFamily productFamily;
+    Product product;
+    Product product2;
+    Component component;
+    Customer customer;
+    Subscription subscription;
+    Customer updatedCustomer;
+    String customerFullName;
+    String updatedCustomerFullName;
+    long lastEventId = 1;
+
+    @BeforeAll
+    void setup() throws IOException, ApiException, InterruptedException {
+        Thread.sleep(3000);
```

Comment:
Why sleep at the beginning?

Reply:
sometimes events are appearing with a little delay. Without it, events caused by execution of other tests could be still arriving during setup